service/controller/key/key.go

```diff
@@ -566,3 +570,68 @@ func vmssInstanceIDBase36(instanceID string) (string, error) {
 
 	return strconv.FormatUint(i, 36), nil
 }
+
+const (
+	ClientIDKey       = "azure.azureoperator.clientid"
+	ClientSecretKey   = "azure.azureoperator.clientsecret"
+	SubscriptionIDKey = "azure.azureoperator.subscriptionid"
+	TenantIDKey       = "azure.azureoperator.tenantid"
+	PartnerIDKey      = "azure.azureoperator.partnerid"
+	defaultAzureGUID  = "37f13270-5c7a-56ff-9211-8426baaeaabd"
+)
+
+func GetTenantAzureClientCredentialsConfig(k8sClient kubernetes.Interface, cr providerv1alpha1.AzureConfig) (auth.ClientCredentialsConfig, error) {
```

Comment:
I think we solely use key functions as similar abstraction as entity class' getters. Usually we don't have any kind of client logic in key package, but I think it would be better to encapsulate it in own package somewhere. Maybe something like `pkg/credential` so these could become `credential.GetTenantAzureClientConfig` and `credential.GetSubscriptionAndPartnerID()` ?

Reply:
Done